com.unity.netcode.gameobjects/Runtime/Messaging/Messages/ParentSyncMessage.cs

```diff
@@ -1,29 +1,49 @@
+using UnityEngine;
+
 namespace Unity.Netcode
 {
     internal struct ParentSyncMessage : INetworkMessage
     {
         public ulong NetworkObjectId;
 
-        public bool IsReparented;
+        public bool WorldPositionStays;
 
         //If(Metadata.IsReparented)
         public bool IsLatestParentSet;
 
         //If(IsLatestParentSet)
         public ulong? LatestParent;
 
+        // Is set when the parent should be removed (similar to IsReparented functionality but only for removing the parent)
+        public bool RemoveParent;
+
+        // These additional properties are used to synchronize clients with the current position,
+        // rotation, and scale after parenting/de-parenting (world/local space relative). This
+        // allows users to control the final child's transform values without having to have a
+        // NetworkTransform component on the child. (i.e. picking something up)
+        public Vector3 Position;
+        public Quaternion Rotation;
+        public Vector3 Scale;
+
         public void Serialize(FastBufferWriter writer)
         {
-            writer.WriteValueSafe(NetworkObjectId);
-            writer.WriteValueSafe(IsReparented);
-            if (IsReparented)
+            BytePacker.WriteValuePacked(writer, NetworkObjectId);
+            writer.WriteValueSafe(RemoveParent);
+            writer.WriteValueSafe(WorldPositionStays);
+            if (!RemoveParent)
             {
                 writer.WriteValueSafe(IsLatestParentSet);
+
                 if (IsLatestParentSet)
                 {
-                    writer.WriteValueSafe((ulong)LatestParent);
+                    BytePacker.WriteValuePacked(writer, (ulong)LatestParent);
```

Comment:
ooh, fancy.

...but might I suggest looking at `BytePacker.WriteValueBitPacked()` and see if it's viable? It may not be. It has better compression than `WriteValuePacked()` but for a ulong it does limit the allowed value to only 61 bits, as it uses the first 3 bits to encode a byte count. (As opposed to `WriteValuePacked()`, which uses a whole byte to encode the byte count on top of the actual value itself, and if all bytes of the value are needed, it actually makes the encoded value larger than the original instead of smaller)

Reply:
(In contrast, `WriteValueBitPacked()` guarantees it will never increase the size of the value, but does have a limitation of only 61 bits... also there's a typo in the comment that says it's limited to 31 bits, but it's not, it's 61.)